app/controllers/dashboard_controller.rb

```diff
@@ -2,11 +2,15 @@
 
 class DashboardController < ApplicationController
   def index
-    @reasons = Reason.all.joins(:posts)
-                     .group('reasons.id')
-                     .select('reasons.*, count(\'posts.*\') as post_count')
-                     .order('inactive ASC, post_count DESC')
-                     .to_a
+    @inactive_reasons, @active_reasons = [true, false].map do |inactive|
+      Reason.all.joins(:posts)
+            .where("reasons.inactive = #{inactive}")
```

Comment:
You've got an SQL injection bug here. String interpolation doesn't perform any parameter sanitization - instead, try `.where(reasons: { inactive: inactive })` or `.where('reasons.inactive = ?', inactive)`.

Reply:
`inactive` is set above on line 5, and it doesn’t contain user input.

Reply:
Heh, so it is. I'm so used to setting variables from params that I didn't notice :)

Reply:
Good call, thanks both. I'll change to `.where('reasons.inactive = ?', inactive)` which is safer, and compatible with pre-existing query style.